pyiron_base/jobs/dynamic.py

```diff
@@ -132,4 +132,53 @@ def _get_template_classes():
     }
 
 
+def create_new_job_type(
+    class_name, input_dict, write_input_funct, collect_output_funct, executable_str
+):
+    from pyiron_base.jobs.job.template import TemplateJob
+
+    def _get_write_input_func(write_input_funct):
+        def write_input(self):
+            write_input_funct(
+                working_directory=self.working_directory,
+                input_dict=self.input.to_builtin(),
+            )
+
+        return write_input
+
+    def _init_constructor_dynamic_func(class_name, input_dict, executable_str):
+        def __init__(self, project, job_name):
+            super(self.__class__, self).__init__(project, job_name)
+            self.__name__ = class_name
```

Comment:
Should be handled already by [this](https://github.com/pyiron/pyiron_base/blob/e359d95f17c17c7489120fe81989fcc35511182c/pyiron_base/jobs/job/generic.py#L130).